### PR TITLE
replace error type assertions with errors.As() in the `destroy` module

### PR DESCRIPTION
**Pull request by Fedosin**

### CODE REVIEW

File: pkg/destroy/openstack/openstack.go [hunk 1]
```diff
@@ -224,7 +224,8 @@ func deleteServers(opts *clientconfig.ClientOpts, filter Filter, logger logrus.F
 		err = servers.Delete(conn, server.ID).ExtractErr()
 		if err != nil {
 			// Ignore the error if the server cannot be found and return with an appropriate message if it's another type of error
-			if _, ok := err.(gophercloud.ErrDefault404); !ok {
+			var gerr gophercloud.ErrDefault404
+			if !errors.As(err, &gerr) {
```
Comment (abhinavdahiya) [inline pkg/destroy/openstack/openstack.go:228]: Since you are not using gerr here in the if block, isn't it better to use errors.Is

Comment (Fedosin) [reply pkg/destroy/openstack/openstack.go:228]: @abhinavdahiya 
Afaik, errors.Is checks if two errors are equal (i.e. err == gerr), but gophercloud generates new errors of `ErrDefault404` type for each request. It means that we have to compare their types (ErrDefault404 or not ErrDefault404).
If you know how I can use errors.Is here, please tell me.

Comment (abhinavdahiya) [reply pkg/destroy/openstack/openstack.go:228]: If you think As as is more appropiate, please explain that in the commit message. i'm fine as long as the reasoning is described.

**Comment by iamemilio** [comment]
can we define var gerr gophercloud.ErrDefault404 as a constant

**Comment by iamemilio** [comment]
> can we define var gerr gophercloud.ErrDefault404 as a constant

lets do it post ff

**Comment by pierreprinetti** [comment]
> can we define var gerr gophercloud.ErrDefault404 as a constant

I'm not sure what you mean... ErrDefault404 is a type:

https://github.com/openshift/installer/blob/8eded0246f38d6a53ff93d92de5e2b0ee937843f/vendor/github.com/gophercloud/gophercloud/errors.go#L130

**Pull request merged**